.github/workflows/sonarcloud.yml

```diff
@@ -1,26 +1,54 @@
+# This GitHub action will checkout and scan third party code.
+# Please ensure that any changes to this action do not perform
+# actions that may result in code from that branch being executed
+# such as installing dependencies or running build scripts.
+
 name: SonarCloud
 
 on:
-  workflow_call:
-    secrets:
-      SONAR_TOKEN:
-        required: true
+  workflow_run:
+    workflows:
+      - Run tests
+    types:
+      - completed
+
+permissions:
+  actions: read
 
 jobs:
   sonarcloud:
+    # Only scan code from non-forked repositories that have passed the tests
+    # This will skip scanning the code for forks, but will run for the main repository on PRs from forks
+    if: ${{ github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.repository.fork == false }}
```

Comment:
It's a bit unclear to me with repository this would come from, in the case of a PR from a fork. Probably this is correct, since this is rooted in a `pull_request` trigger that should be in the context of our repo. But either way, this is a safe way to test, we can follow up if we need to 👍